src/Tempest/Http/src/Route.php

```diff
@@ -55,4 +44,29 @@ public function setHandler(MethodReflector $handler): self
 
         return $this;
     }
+
+    /** @return string[] */
+    public static function getRouteParams(string $uriPart): array
+    {
+        $regex = '#\{'. self::ROUTE_PARAM_NAME_REGEX . self::ROUTE_PARAM_CUSTOM_REGEX .'\}#';
+
+        preg_match_all($regex, $uriPart, $matches);
+
+        return $matches[1] ?? [];
+    }
+
+    /**
+     * Splits the route URI into separate segments
+     *
+     * @example '/test/{id}/edit' becomes ['test', '{id}', 'edit']
+     * @return string[]
+     */
+    public function split(): array
+    {
+        $parts = explode('/', $this->uri);
+
+        return array_values(
+            array_filter($parts, static fn (string $part) => ! empty($part))
```

Comment:
I think we should use `=== ''` instead of `empty`, because `empty('0')` is `true`. So if I understand correctly, a route like `api/0/foo` would become `['api', 'foo']` instead of `['api', '0', 'foo']`.

```suggestion
            array_filter($parts, static fn (string $part) => $part !== '')
```

This will most likely be a corner case but I think tis will happen in the wild for api versionning or similar things.

Reply:
Yes, you correct! thank you! I will, use your suggestion as is to ensure your contribution and  add a quick test to proof it  in tests.